apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/SwapWebView.tsx

```diff
@@ -77,6 +88,52 @@ const SwapWebView = ({ swapState, redirectToProviderApp }: SwapWebProps) => {
         onSwapWebviewError(params);
         return Promise.resolve();
       },
+      "custom.saveSwapToHistory": ({
+        params,
+      }: {
+        params: { swap: SwapProps; transaction_id: string };
+      }) => {
+        const { swap, transaction_id } = params;
+        if (!swap || !transaction_id || !swap.provider || !swap.fromAmount || !swap.toAmount) {
+          return Promise.reject("Cannot save swap missing params");
+        }
+        const fromId = getAccountIdFromWalletAccountId(swap.fromAccountId);
+        const toId = getAccountIdFromWalletAccountId(swap.toAccountId);
+        if (!fromId || !toId) return Promise.reject("Accounts not found");
+        const operationId = `${fromId}-${transaction_id}-OUT`;
+
+        const swapOperation: SwapOperation = {
+          status: "pending",
+          provider: swap.provider,
+          operationId,
+          swapId: transaction_id,
+          receiverAccountId: toId,
+          tokenId: toId,
+          fromAmount: new BigNumber(swap.fromAmount),
+          toAmount: new BigNumber(swap.toAmount),
+        };
+
+        dispatch(
+          updateAccountWithUpdater(fromId, account => {
+            const fromCurrency = getAccountCurrency(account);
+            const isFromToken = fromCurrency.type === "TokenCurrency";
+            const subAccounts = account.type === "Account" && account.subAccounts;
+            return isFromToken && subAccounts
+              ? {
+                  ...account,
+                  subAccounts: subAccounts.map<SubAccount>((a: SubAccount) => {
+                    const subAccount = {
+                      ...a,
+                      swapHistory: [...a.swapHistory, swapOperation],
+                    };
+                    return a.id === fromId ? subAccount : a;
```

Comment:
This function is taken from https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/exchange/swap/addToSwapHistory.ts#L47

Where swaps are saved to subAccounts. I don't why but I think It might be important

Reply:
All good, happy to leave it be in that case 👍